hibernate-core/src/main/java/org/hibernate/internal/util/xml/XMLHelper.java

```diff
@@ -22,23 +23,34 @@
 public final class XMLHelper {
 	private final DocumentFactory documentFactory;
 
-	public XMLHelper(ClassLoaderService classLoaderService) {
```

Comment:
Do Hibernate Search or other Hibernate projects new up XMLHelper via XMLHelper(ClassLoaderService classLoaderService)?

Reply:
Overall, the change looks good to me.

Reply:
If there is concern about existing references to this internal class from other Hibernate projects, we could add XMLHelper(ClassLoaderService) ctor back in.